src/NHSD.BuyingCatalogue.Identity.Api/Controllers/ErrorController.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Threading.Tasks;
+using IdentityServer4.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace NHSD.BuyingCatalogue.Identity.Api.Controllers
+{
+    public class ErrorController : Controller
+    {
+        private readonly IIdentityServerInteractionService _interactionService;
+        private readonly ILogger<ErrorController> _logger;
+
+        public ErrorController(
+            IIdentityServerInteractionService interactionService,
+            ILogger<ErrorController> logger)
+        {
+            _interactionService = interactionService;
+            _logger = logger;
+        }
+
+        public async Task<IActionResult> Index(string errorId)
+        {
+            var message = await _interactionService.GetErrorContextAsync(errorId);
+            _logger.Log(LogLevel.Error, "Request for client {0} failed with error code {1}: {2}",
```

Comment:
One suggestion here is to call _logger.LogInformation rather than the specific implementation to keep it consistent with the rest of the product suite. Also review the use of passing the message into the logger as an object. This way Serilog can get involved & JSON stringify the data. I'm not quite sure of the syntax here, hence the request for review.